IEP-1667 support esp-idf-size 2.0 and use idf.py size command

sigmaaa opened the pull request:
## Description

Added pip show check for esp-idf-size package

Fixes # ([IEP-1667](https://jira.espressif.com:8443/browse/IEP-1667))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Install the the latest esp-idf 6.0, and make sure it includes the esp-idf-size package with at least version 2.0. Check it from the esp-idf venv Python.
Then check the application size analysis.

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Component 1
- Component 2

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added archive and file-specific size/symbol queries and richer memory breakdowns shown in the size viewer.

* **Bug Fixes**
  * Improved handling of missing/invalid size output to show a clear error message instead of failing; NaN values are now treated as zero.

* **Documentation**
  * Added user-facing messages/localization for no-data and error states.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kolipakakondal commented on bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeDataManager.java:
```diff
@@ -305,4 +323,34 @@ protected JSONObject getJSON(String jsonOutput)
+	{
+		if (pipShowOutput == null || pipShowOutput.isBlank())
+		{
+			return false;
```
Add logging If pip show fails and pipShowOutput is null or blank.

kolipakakondal commented on bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/size/IDFSizeDataManager.java:
```diff
@@ -64,7 +64,10 @@ public List<LibraryMemoryComponent> getDataList(IFile mapFile) throws Exception
+		boolean useNg = isNgArgSupported(pipShowOutput);
+		List<String> commandArgs = getCommandArgs(pythonExecutablePath, mapFile, useNg);
 		String detailsJsonOp = getOutput(mapFile, commandArgs);
 		detailsJsonOp = detailsJsonOp.replace("NaN", "0"); //$NON-NLS-1$ //$NON-NLS-2$
```
detailsJsonOp can be null

kolipakakondal reviewed (changes requested):
Hi @sigmaaa Thanks for the PR. Could you take a look at the PR comments? Otherwise, it looks good to me.

sigmaaa commented:
Hi @kolipakakondal,

Thanks for the review. I’ve completely changed the approach in the latest commit and switched to using idf.py size --format instead. Please take another look. I’ve also added some guard checks to prevent the UI from breaking in case of unexpected output from the idf.py size command.
<img width="1226" height="674" alt="image" src="https://github.com/user-attachments/assets/28fac9f2-51a0-4756-bb9a-bc60bec2931c" />

kolipakakondal reviewed (approved):
LGTM
